src/components/search/data/searchLoader.js

```diff
@@ -18,11 +19,19 @@ export default function makeSearchLoader(queryClient) {
       authenticatedUser,
       enterpriseSlug,
     });
-    const searchData = [
-      queryClient.ensureQueryData(
-        queryAcademiesList(enterpriseCustomer.uuid),
-      ),
-    ];
+    const academies = await queryClient.ensureQueryData(queryAcademiesList(enterpriseCustomer.uuid));
```

Comment:
This academies list query should remain in the `searchData` array to ensure that the API call to fetch academies is made in parallel with the API call to fetch highlight sets below (for performance, API calls made serially when not necessary should be avoided).

From a related [comment](https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/1054#discussion_r1565824713) on the previous review:

> you can access that data pre-seeded in the query cache by using `queryClient.getQueryData` ([docs](https://tanstack.com/query/v4/docs/reference/QueryClient/#queryclientgetquerydata)) after the `await Promise.all(searchData)` to determine whether the customer has just 1 academy returned.

Example:

```jsx
const academiesListQuery = queryAcademiesList(enterpriseCustomer.uuid);
// ...
await Promise.all(searchData); // searchData should continue to contain both academies and highlight sets queries
const academies = queryClient.getQueryData(academiesListQuery.queryKey); // due to the `.ensureQueryData` call above, at this point in the code, you can assume `academies` data exists in the query cache and access it directly with `getQueryData`).
if (enterpriseCustomer.enableOneAcademy && academies.length === 1) {
  // ...
  return redirect(redirectPath);
}
```